model/host/db.go

```diff
@@ -130,6 +137,45 @@ func AllIdleEphemeral() ([]Host, error) {
 	return Find(query)
 }
 
+// IdleEphemeralGroupedByDistroId groups and collates the following grouped and ordered by {distro.Id: 1}:
+// - []host.Host of ephemeral hosts without containers which having no running task, ordered by {host.CreationTime: 1}
+// - the total number of ephemeral hosts with status: evergreen.HostRunning
+func IdleEphemeralGroupedByDistroId() ([]IdleHostsByDistroID, error) {
+	var idlehostsByDistroID []IdleHostsByDistroID
+	pipeline := []mgobson.M{
+		{
+			"$match": mgobson.M{
+				StartedByKey:     evergreen.User,
+				StatusKey:        evergreen.HostRunning,
+				ProviderKey:      mgobson.M{"$in": evergreen.ProviderSpawnable},
+				HasContainersKey: mgobson.M{"$ne": true},
```

Comment:
seems ok to leave this out.

Reply:
In that case, should I also remove the `HasContainersKey` predicate from the `AllIdleEphemeral` function?

Reply:
lets leave this as is

Reply:
OK